torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -261,18 +263,17 @@ BestEffortReplay::BestEffortReplay(
           replay_id2expr_map.find(id) == replay_id2expr_map.end(),
           "Error trying to map rfactor root domain during replay.",
           " An IterDomain was found to be used in more than one expression.");
-      // Only want to forward rfactor in map
-      replay_id2expr_map[id] = replay_expr;
 
-      auto out_ids = ir_utils::filterByType<IterDomain>(replay_expr->outputs());
+      replay_id2expr_map[id] = replay_expr;
+    }
 
-      if (std::any_of(out_ids.begin(), out_ids.end(), [](IterDomain* id) {
-            return id->isRFactorProduct();
-          })) {
-        auto inp_ids =
-            ir_utils::filterByType<IterDomain>(replay_expr->inputs());
-        replay_rfactor_ids.insert(inp_ids.begin(), inp_ids.end());
-      }
+    // Only want to forward rfactor in map
+    auto out_ids = ir_utils::filterByType<IterDomain>(replay_expr->outputs());
+    if (std::any_of(out_ids.begin(), out_ids.end(), [](IterDomain* id) {
+          return id->isRFactorProduct();
+        })) {
+      auto inp_ids = ir_utils::filterByType<IterDomain>(replay_expr->inputs());
+      replay_rfactor_ids.insert(inp_ids.begin(), inp_ids.end());
```

Comment:
Moved out of the innermost loop as it doesn't depend on the loop of `replay_expr->inputs()`.